Automattic/wp-calypso
# Checkout: Add the “Number of licenses” dropdown in Akismet checkout

In this PR we add the "Number of licenses" dropdown in Akismet checkout, along with the logic & functionality to adjust the Akismet Pro 500 product's licensed-quantity value.  (see image and project links below)

![Markup 2023-11-28 at 18 19 04](https://github.com/Automattic/wp-calypso/assets/11078128/d534015e-9164-460a-9bd1-f891180e1468)




**Related to:**
- PT: Akismet Checkout – Multiple licenses: pbNhbs-8My-p2
- Akismet Checkout: Multiple Licenses – Implementation Plan: pbNhbs-8P6-p2

**Still TODO in a follow-up PR(s):**
1. Add "Unlimited sites (Akismet Business)" as an option in the dropdown, that when “Unlimited sites (Akismet Business)” is selected in the dropdown, replace the Pro product in the cart with Akismet Business. If 1 - 4 licenses is selected, replace the Business product with Akismet Pro.
2. Adjust the line-item cost/price display per per project/design specs. (Or possibly adjust the line-item to display the math, for example: "3 licenses x $119.40 per license = $358.20”
3. Reflect the number of licenses selected within the "Included with your purchase" summary in the right side column.
4. ~~Possibly refactor the dropdown into dropdown-like buttons (like the billing-term variation picker) as opposed to an HTML Select element (because the default styles and behavior of an HTML Select element is possibly undesired (especially in mobile view) in checkout).~~


## Testing Instructions

- First apply patch D128326-code to your sandbox and sandbox `public-api.wordpress.com`.
- Checkout this PR branch and `yarn-start`
- Go to https://akismet.com/pricing and select/click the  "Get Pro" button. 
- On the checkout page, in the URL, replace `https://wordpress.com` with `http://calypso.localhost:3000` and press enter.
- You should see the "Number of licenses" dropdown, like in the image above.
- Select various options in the dropdown, verify it adjusts the number of licenses and adjusts the price accordingly.
- Change billing-term variation and verify everything still works as intended.
- Also try selecting the number of sites using the keyboard. (`Enter` or `space` opens the dropdown. `up` and `down` arrows to change the quantity, `Enter` to select.)
- Try to break things or find issues.


## Pre-merge Checklist
